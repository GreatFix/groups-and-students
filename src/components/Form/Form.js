import React from 'react'
import classes from './Form.module.css'
import Button from '../../components/Button/Button'
import PropTypes from 'prop-types'

const Form = (props) => {
  return (
    <form onSubmit={props.onSubmit} className={classes.Form}>
      {props.children}
      <div className={classes.Submit}>
        <Button type={'submit'} color={'success'}>
          {props.submitText}
        </Button>
      </div>
    </form>
  )
}

Form.propTypes = {
  submitText: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  onSubmit: PropTypes.func.isRequired,
}

Form.defaultProps = {
  submitText: '',
  children: null,
  onSubmit: null,
}

export default React.memo(Form)
