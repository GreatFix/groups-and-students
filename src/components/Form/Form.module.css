.Submit {
  margin: 2px;
  padding: 5px;
  text-align: center;
}
