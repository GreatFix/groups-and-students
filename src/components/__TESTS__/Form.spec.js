import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Form from '../Form/Form'

const TEST_TEXT = 'TEST_TEXT'
const TEST_BUTTON_TEXT = 'TEST_BUTTON_TEXT'
const TEST_VALUE = 'TEST_VALUE'

describe('Form', () => {
  test('renders a children', () => {
    const { getByDisplayValue } = render(
      <Form submitText={TEST_BUTTON_TEXT} onSubmit={() => {}}>
        <input defaultValue={TEST_TEXT} />
      </Form>
    )
    expect(getByDisplayValue(TEST_TEXT)).toBeInTheDocument()
  })

  test('render submitText', () => {
    const { getByText } = render(
      <Form submitText={TEST_BUTTON_TEXT} onSubmit={() => {}}>
        <input defaultValue={TEST_TEXT} />
      </Form>
    )
    const submit = getByText(TEST_BUTTON_TEXT)
    expect(submit).toHaveAttribute('type', 'submit')
  })

  test('call onSubmit', () => {
    const mockOnSubmit = jest.fn((e) => e.target.value)
    const { getByText } = render(
      <Form submitText={TEST_BUTTON_TEXT} onSubmit={mockOnSubmit}>
        <input defaultValue={TEST_TEXT} />
      </Form>
    )
    const submit = getByText(TEST_BUTTON_TEXT)
    fireEvent.submit(submit, {
      target: {
        value: TEST_VALUE,
      },
    })

    expect(mockOnSubmit.mock.calls.length).toBe(1)
    expect(mockOnSubmit.mock.results[0].value).toBe(TEST_VALUE)
  })

  test('render properly', () => {
    const { container } = render(
      <Form submitText={TEST_BUTTON_TEXT} onSubmit={() => {}}>
        <input defaultValue={TEST_TEXT} />
      </Form>
    )
    expect(container).toMatchSnapshot()
  })
})
