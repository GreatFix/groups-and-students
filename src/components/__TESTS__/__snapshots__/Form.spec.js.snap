// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form render properly 1`] = `
<div>
  <form
    class="Form"
  >
    <input
      value="TEST_TEXT"
    />
    <div
      class="Submit"
    >
      <button
        class="Button "
        style="background-color: rgb(64, 200, 30);"
        type="submit"
      >
        TEST_BUTTON_TEXT
      </button>
    </div>
  </form>
</div>
`;
